'Window' ordering ability

Louis-Lesage opened the pull request:
I've written some code that should fix #100 by adding the ability to switch the order of windows at run time using the arrows. Now you can choose which window you want to see when you can only see one or 2.

Louis-Lesage commented:
I will be fixing the errors and updating the changelog shortly.

Louis-Lesage commented:
That's a really good idea. I'll implement this right away!

imsnif commented on src/display/components/help_text.rs:
```diff
@@ -8,18 +8,24 @@ pub struct HelpText {
+                TEXT_WHEN_NOT_PAUSED.to_owned()
             };
 
+            if rect.width >= FIRST_WIDTH_BREAKPOINT {
```
This is a really nice touch

imsnif commented on src/main.rs:
```diff
@@ -205,6 +214,10 @@ where
                                 display_handler.unpark();
                             }
+                            Event::Key(Key::Char('\t')) => {
+                                ui_offset.fetch_add(1, Ordering::SeqCst);
```
I'm not thrilled about this... The issue here is that this number will grow infinitely. It's not terrible (we have much larger numbers in the app), but I would rather not do it this way. What do you think of cloning the UI here and adding a toggle method to it? Or maybe you have another idea?

Louis-Lesage replied:
I could try to bound the value of the `ui_offset` by using modulo and the number of windows (3 here, but maybe it could change?).

imsnif replied:
That's also a possibility, definitely. Do you think you can get this number from a clone of the ui? that way we won't have to worry about changing this when that number changes, as you say.

imsnif commented on src/main.rs:
```diff
@@ -205,6 +214,10 @@ where
                             }
+                            Event::Key(Key::Char('\t')) => {
+                                ui_offset.fetch_add(1, Ordering::SeqCst);
+                                display_handler.unpark();
```
This causes a bug: try to download something big and then open the app and spam TABs. What would happen is that you'll see the download speed going down. The reason is that we refresh the UI after every tab, and since the display thread is also in charge of measuring the bandwidth (in units of 1 second), it throws the calculation off.

I see why you did this, and of course also realize this happens with the pause too (but because of usability, this is less likely to happen with pause).

We should probably add some sort of "refresh ui without changing state" method. But that can be done separately.

Louis-Lesage commented:
I've been getting a lot of assignments lately might take a while for me to look at the issues!

imsnif commented:
No worries - take your time

Louis-Lesage commented:
Sorry it took so long, i had a lot of work on my shoulders during this trimester. 
Anyway here is what i did regarding the changes you requested.
Let me know what you think!

imsnif commented:
Hey @Louis-Lesage - no worries at all. I'm happy to see you picked this up again.

I plan on going over this during the weekend, but I do have one request: do you think it would be possible for you to do a git rebase so that this PR will only show your changes and not everything that happened in master? It will make it much easier for me to review. Thanks!

Louis-Lesage commented:
Here you go @imsnif! I've updated what you wanted regarding the `dns_content` and `tab_text` and it looks great. I've added some test for it too. 
Let me know if i missed anything!
